pkg/rc/replication_controller.go

```diff
@@ -96,6 +99,16 @@ type consulStore interface {
 	NewUnmanagedSession(session, name string) consul.Session
 }
 
+// nodeTransfer encapsulates the information required to perform and cancel a node transfer
+// off an ineligible node
+type nodeTransfer struct {
+	newNode       types.NodeName
+	oldNode       types.NodeName
+	quit          chan struct{}
+	sessionCancel func()
```

Comment:
nit: use `context.CancelFunc` (which is an alias for `func()` but makes it clearer what it's for)